### DISCUSSION TITLE
Accessing a collection using front matter variable as tag

### DISCUSSION THREAD

**Discussion by sarahgibeau** [11ty/eleventy#2421] [Q&A]
I'm trying to create tags based on on a value from front matter and then filter by that variable, but I can't even get the latter part to work. Say my front matter looks like:
---
tags: example
name: example
---
I would like to be able to do something like:

{% for item in collections[name] %} ...

However nothing shows up. I've tried every variation I can think of with quotes or not etc, what am I missing? I'm using nunjucks. Also, is it possible to assign a front matter variable as a tag? I have my tags in a json file for each directory, could I do something like:

tags: {{name}}

? 
Thank you!

**Comment by pdehaan** [comment]
This works for me:

```njk
---
# src/pages/index.njk
title: InDeX
tags:
  - example
name: example
---

{% for item in collections[name] %}
<p>title={{ item.data.title }}; {{ item.url }}</p>
{% endfor %}
```

### src/pages/pages.json
This will apply the "example" tag to each of the pages in the src/pages/* directory (and subdirectories)
```json
{
  "tags": ["example"]
}
```

### src/pages/one.njk
Plus I also have a src/pages/two.njk and src/pages/three.njk with similar content, where only the `title` frontmatter is changed.
```njk
---
# src/pages/one.njk
title: OnE
---
<h1>{{ title }}</h1>
```

### OUTPUT
```html
<p>title=OnE; /pages/one/</p>
<p>title=tWo; /pages/two/</p>
<p>title=ThReE; /pages/three/</p>
<p>title=InDeX; /pages/</p>
```

**Reply by pdehaan** [reply]
Setting dynamic tags is _possible_, but in my experience, it doesn't auto-populate collections like hardcoded tags do. So you might have to manually create collections in your .eleventy.js config file based on globs, tags, frontmatter, etc.

**Reply by pdehaan** [reply]
Pushed my repo to https://github.com/pdehaan/11ty-2421

**Reply by pdehaan** [reply]
I also put this in an index.njk file which should log out your collection names and number of pages:

```njk
{%- for name, pages in collections %}
- "{{ name }}" => {{ pages.length }} pages
{%- endfor %}
```

### OUTPUT
```html
- "all" => 5 pages
- "example" => 4 pages
```
